docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Async Processing
+---
+import {Icons} from '/src/components/icons'
+
+:::info
+
+This page covers the Responsive Async Processor: its benefits, its limitations, how to use it, and how to tune it.
+
+The Async Processor is available with Responsive client **v0.24.0** onward.
+:::
+
+## Why use the Async Processor
+Async processing allows Kafka Streams applications to concurrently process records within a single task, thus changing the upper limit of parallelization from the level of a Kafka partition to the level of a record key. 
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+While async processing can help improve performance of any application with a large keyspace, it was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
+
+This makes the Responsive Async Processor the perfect tool to complement Responsive state stores, meaning you don’t need to sacrifice performance in order to gain the availability and elasticity benefits of remote storage with Kafka Streams!
+
+## Expected Performance Gains
+
+As with all real-world systems, performance depends heavily on the specific use case and parameters. However, in all of our experiments, async processing showed an improvement over regular Kafka Streams when a topology includes a processor which makes a remote call. 
```

Comment:
Maybe "which makes a remote or long-blocking call"? Technically people can have very long running processors that don't interact with remote systems -- though that's rare. But is it only rare because, like RPCs, it doesn't work very well with regular Kafka Streams? 🤔

Reply:
good call. will update.